### PR TITLE
Documentation update for Optuna (alpha release)

### PR DESCRIPTION
**Pull request by nv-braf**
Add documentation about new search mode, CLI/Config options, and example output.

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -227,6 +227,27 @@ cpu_only_composing_models: <comma-delimited-string-list>
 # Enables the searching of request rate (instead of concurrency)
 [ request_rate_search_enable: <bool> | default: false]
 
+# Minimum percentage of the search space to profile when using Optuna
+[ min_percentage_of_search_space: <int> | default: 5]
+
+# Maximum percentage of the search space to profile when using Optuna
+[ max_percentage_of_search_space: <int> | default: 10]
```
Comment (tgerdesnv) [inline docs/config.md:234]: 5-10% seems like a very small band. Do your experiments show that <5% is generally bad and >10% is overkill?

File: docs/config.md [hunk 2]
```diff
@@ -227,6 +227,27 @@ cpu_only_composing_models: <comma-delimited-string-list>
...
+
+# Minimum number of trials to profile when using Optuna
+[ optuna_min_trials: <int> | default: 20]
```
Comment (tgerdesnv) [inline docs/config.md:237]: One of [percentage vs trials] has precedence, right? And the other is only used if it is specified? If so, is there any point in specifying a default?  Reading the docs it isn't clear to me what will be searched by default

File: docs/config.md [hunk 3]
```diff
@@ -227,6 +227,27 @@ cpu_only_composing_models: <comma-delimited-string-list>
...
+
+# Maximum number of trials to profile when using Optuna
+[ optuna_max_trials: <int> | default: 200]
+
+# Number of trials without improvement before triggering early exit when using Optuna
+[ optuna_early_exit_threshold: <int> | default: 10]
+
+# Use the concurrency formula instead of searching the concurrency space in Optuna search mode
```
Comment (tgerdesnv) [inline docs/config.md:245]: Can the formula be specified or linked here?

File: docs/config_search.md [hunk 4]
```diff
@@ -276,6 +280,106 @@ profile_models:
...
+
+```yaml
+model_repository: /path/to/model/repository/
+
+run_config_search_mode: optuna
+profile_models:
+  - model_A
+```
+
+A number of new configuration options were added to support tailoring the Optuna search to your needs:
```
Comment (tgerdesnv) [inline docs/config_search.md:314]: Looking at the actual rendered page, this is part of the section above, but it seems related to the thing below it, not the thing above it. I think the horizontal bars need to be tweaked

File: docs/config_search.md [hunk 5]
```diff
@@ -276,6 +280,106 @@ profile_models:
...
+        max_queue_delay_microseconds: [100, 200, 300]
+    parameters:
+      batch_sizes: 1, 2, 4, 8, 16
+```
+
+_The debug output showing how the space will be searched:_
+
+```yaml
+Number of configs in search space: 720
+   batch_sizes: [1, 2, 4, 8, 16] (5)
```
Comment (tgerdesnv) [inline docs/config_search.md:352]: Is there an example that isn't all the options we already support?

File: docs/config_search.md [hunk 6]
```diff
@@ -276,6 +280,106 @@ profile_models:
...
+Trial 7 of 200:
+  Creating model config: model_A_config_6
+  Setting dynamic_batching to {'max_queue_delay_microseconds': 200}
+  Setting instance_group to [{'count': 4, 'kind': 'KIND_GPU'}]
+  Setting max_batch_size to 64
+
+  Profiling model_A_config_6: client batch size=4, concurrency=256
+  Objective score for model_A_config_6: 57 --- Best: model_A_config_4 (83)
+```
+
```
Comment (tgerdesnv) [inline docs/config_search.md:382]: We have all these thick solid bars between subsections, and then nothing between the end of this feature and the start of the next major feature (Ensemble)

File: model_analyzer/config/generate/optuna_plus_concurrency_sweep_run_config_generator.py [hunk 7]
```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
```
Comment (tgerdesnv) [inline model_analyzer/config/generate/optuna_plus_concurrency_sweep_run_config_generator.py:1]: Is this branch properly based? There are a ton of non-documentation changes in this PR.

File: docs/config.md (hunk 2 above)
Comment (nv-braf) [reply docs/config.md:237]: Right now I made percentage have precedence. So, no there's not really a point of specifying a default for the other.

File: docs/config_search.md (hunk 5 above)
Comment (nv-braf) [reply docs/config_search.md:352]: This example uses options that quick search doesn't support (and has enumerated lists).

File: model_analyzer/config/generate/optuna_plus_concurrency_sweep_run_config_generator.py (hunk 7 above)
Comment (nv-braf) [reply model_analyzer/config/generate/optuna_plus_concurrency_sweep_run_config_generator.py:1]: I needed to rebase the parent branch with main (to fix grpc issues) and now I have a ton of merge conflicts on rebase that I can't easily figure out how to resolve. I was planning on remaking this branch, and sending it out for review before anyone commented, but you beat me to it.

Any suggestions on how I can fix this without using `git rebase optuna-search-mode`?

File: docs/config_search.md (hunk 4 above)
Comment (nv-braf) [reply docs/config_search.md:314]: I followed the same formatting as all the other sections. It looks fine to me when it's rendered.

**Pull request closed without merging**